### PR TITLE
Migrate all the project files to the new .NET Core csproj format et. al.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
For compatibility, they _still_ target .NET Framework 4.5.1.
Also, ignore files from Rider, move the solution file to the project's directory root (as is usually done), and add the word "Scriban" to Resharper's spell-check dictionary for this solution.

### CODE REVIEW

File: scriban.sln [hunk 1]
```diff
@@ -3,32 +3,36 @@ Microsoft Visual Studio Solution File, Format Version 12.00
 # Visual Studio 15
```
Comment (xoofx) [inline scriban.sln]: Why the file was moved from src to top level? I would like to keep it where it was.

**Comment by xoofx** [comment]
> Also, ignore files from Rider, move the solution file to the project's directory root (as is usually done),

For this, I would like that you revert this change. I have been adopting the sln in the src folder to keep the top folder smaller with only the required files.

File: scriban.sln (hunk 1 above)
Comment (teo-tsirpanis) [reply scriban.sln]: Done.

**Comment by xoofx** [comment]
Thanks!

**Pull request merged**